api/http_lifecycle.go

```diff
@@ -0,0 +1,160 @@
+// NOTE: Subject to change, do not rely on this package from outside git-lfs source
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// HttpLifecycle serves as the default implementation of the Lifecycle interface
+// for HTTP requests. Internally, it leverages the *http.Client type to execute
+// HTTP requests against a root *url.URL, as given in `NewHttpLifecycle`.
+type HttpLifecycle struct {
+	// root is the root of the API server, from which all other sub-paths are
+	// relativized
+	root *url.URL
+	// client is the *http.Client used to execute these requests.
+	client *http.Client
+	// authenticateRequests stores whether or not the HttpLifecycle should
+	// authenticate its HTTP requests
+	authenticateRequests bool
+}
+
+var _ Lifecycle = new(HttpLifecycle)
+
+// NewHttpLifecycle initializes a new instance of the *HttpLifecycle type with a
+// new *http.Client, and the given root (see above).
+func NewHttpLifecycle(root *url.URL) *HttpLifecycle {
+	return &HttpLifecycle{
+		root:   root,
+		client: new(http.Client),
```

Comment:
See `httputil.NewHttpClient` for some cases you won't be handling by just creating a standard `http.Client`.